driver-core/src/test/unit/com/mongodb/client/model/changestream/ChangeStreamDocumentCodecSpecification.groovy

```diff
@@ -56,30 +57,37 @@ class ChangeStreamDocumentCodecSpecification extends Specification {
                 new ChangeStreamDocument<Document>(
                         BsonDocument.parse('{token: true}'),
                         BsonDocument.parse('{db: "databaseName", coll: "collectionName"}'),
+                        null,
                         Document.parse('{key: "value for fullDocument"}'),
                         new BsonDocument('_id', new BsonInt32(1)),
                         new BsonTimestamp(1234, 2),
                         OperationType.INSERT,
-                        null
-                ),
+                        null),
                 new ChangeStreamDocument<Document>(
                         BsonDocument.parse('{token: true}'),
                         BsonDocument.parse('{db: "databaseName", coll: "collectionName"}'),
+                        null,
                         Document.parse('{key: "value for fullDocument"}'),
                         new BsonDocument('_id', new BsonInt32(2)),
                         null,
                         OperationType.UPDATE,
-                        new UpdateDescription(['a', 'b'], BsonDocument.parse('{c: 1}'))
-                )
+                        new UpdateDescription(['a', 'b'], BsonDocument.parse('{c: 1}'))),
+                new ChangeStreamDocument<Document>(
+                        BsonDocument.parse('{token: true}'),
+                        BsonDocument.parse('{db: "databaseName", coll: "collectionName"}'),
+                        BsonDocument.parse('{db: "databaseName2", coll: "collectionName2"}'),
+                        null, null, null, OperationType.RENAME, null),
         ]
-        clazz << [Document, Document]
+        clazz << [Document, Document, Document]
         json << [
             '''{_id: {token: true}, ns: {db: "databaseName", coll: "collectionName"}, documentKey : {_id : 1},
                 fullDocument: {key: "value for fullDocument"}, clusterTime: { "$timestamp" : { "t" : 1234, "i" : 2 } }
                 operationType: "insert"}''',
             '''{_id: {token: true}, ns: {db: "databaseName", coll: "collectionName"}, documentKey : {_id : 2},
                 fullDocument: {key: "value for fullDocument"},
                 operationType: "update", updateDescription: {removedFields: ["a", "b"], updatedFields: {c: 1}}}''',
+            '''{_id: {token: true}, ns: {db: "databaseName", coll: "collectionName"},
+                to: {db: "databaseName2", coll: "collectionName2"}, operationType: "rename"}''',
```

Comment:
The docs has a full event document, can we add a test to test we can parse all of that:

```
{
   _id : { <BSON Object> },
   "operationType" : "<operation>",
   "fullDocument" : { <document> },
   "ns" : {
      "db" : "<database>",
      "coll" : "<collection"
   },
   "to" : {
      "db" : "<database>",
      "coll" : "<collection"
   },
   "documentKey" : { "_id" : <value> },
   "updateDescription" : {
      "updatedFields" : { <document> },
      "removedFields" : [ "<field>", ... ]
   }
   "clusterTime" : <Timestamp>,
   "txnNumber" : <NumberLong>,
   "lsid" : {
      "id" : <UUID>,
      "uid" : <BinData>
   }
}
```
https://docs.mongodb.com/manual/reference/change-events/#change-stream-output

Alternatively, the docs gives examples of each event type..

Reply:
Great idea. Helped me find all the nullable things in an unmodified change stream document.   

Done.